src/webextension/background/authorization.js

```diff
@@ -0,0 +1,214 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import jose from "node-jose";
+
+const OAUTH_CLIENT_ID = "6b7245cb84132a90";
+const OAUTH_CLIENT_SECRET = "bc56d58f7996e5f40db8e1c05f61e745cd4e0664cb4d1f3f787667bcca3e4e79";
+const OAUTH_REDIRECT_URI = "https://2aa95473a5115d5f3deb36bb6875cf76f05e4c4d.extensions.allizom.org/";
+const OAUTH_URL_PREFIX = "https://oauth-stable.dev.lcip.org/v1";
+const OAUTH_PROFILE_URI = "https://stable.dev.lcip.org/profile/v1";
+const OAUTH_SCOPES = ["profile"];
+const FXA_URI_PREFIX = "https://stable.dev.lcip.org/auth/v1";
+
+function toQuery(params) {
+  let query = Object.keys(params).map((k) => {
+    let v = params[k];
+    return `${encodeURIComponent(k)}=${encodeURIComponent(v)}`;
+  });
+  return query.join("&");
+}
+function fromQuery(query) {
+  let params = {};
+  query.substring(1).split("&").forEach((p) => {
+    let [k, v] = p.split("=");
+    params[decodeURIComponent(k)] = decodeURIComponent(v || "");
+  });
+  return params;
+}
+
+async function makeAuthzRequest(state, challenge) {
+  let code;
+  code = (new TextEncoder()).encode(challenge);
+  code = await jose.JWA.digest("SHA-256", code);
+  code = jose.util.base64url.encode(code);
+
+  let query = toQuery({
+    "client_id": OAUTH_CLIENT_ID,
+    "state": state,
+    "redirect_uri": OAUTH_REDIRECT_URI,
+    "scope": OAUTH_SCOPES.join(" "),
+    "access_type": "offline",
+    "code_challenge": code,
+    "code_challenge_method": "S256",
+  });
+
+  return `${OAUTH_URL_PREFIX}/authorization?${query}`;
+}
+function parseAuthzResponse(url, state) {
+  url = new URL(url);
+
+  let search = url.search;
+  if (!search) {
+    throw new Error("OAUTH response parameters missing");
+  }
+
+  let params = fromQuery(search);
+  if (params.state !== state) {
+    throw new Error("OAUTH state does not match");
+  }
+
+  return params;
+}
+
+async function requestIt(reason, url, config) {
+  let response = await fetch(url, config);
+  let body = await response.json();
+  if (!response.ok) {
+    // body is FxA error response
+    console.log(`${reason} request failed: ${JSON.stringify(body)}`);
+    let err = new Error(`${reason} request failed: ${body.message}`);
+    Object.assign(err, {
+      code: err.code,
+      errno: err.errno,
+    });
```

Comment:
What does this do? It looks like you're just doing `err.code = err.code; err.errno = err.errno`.